src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -446,81 +446,20 @@ export class AttachToDropdown extends SelectBox {
-					connections.unshift(msgSelectConnection);
-				}
-				else {
-					if (!find(connections, x => x === msgAddNewConnection)) {
-						connections.push(msgAddNewConnection);
-					}
-				}
-				this.setOptions(connections, 0);
-			}
-		}
-	}
-
-	private loadWithSelectConnection(connections: string[]): void {
-		if (connections && connections.length > 0) {
-			if (!find(connections, x => x === msgSelectConnection)) {
-				connections.unshift(msgSelectConnection);
-			}
-
-			if (!find(connections, x => x === msgAddNewConnection)) {
-				connections.push(msgAddNewConnection);
+			if (!find(connections, x => x === msgChangeConnection)) {
+				connections.push(msgChangeConnection);
 			}
 			this.setOptions(connections, 0);
-		}
-	}
-
-	//Get connections from context
-	public getConnections(model: INotebookModel): string[] {
-		let otherConnections: ConnectionProfile[] = [];
-		model.contexts.otherConnections.forEach((conn) => { otherConnections.push(conn); });
-		// If current connection connects to master, select the option in the dropdown that doesn't specify a database
-		if (!model.contexts.defaultConnection.databaseName) {
-			this.selectWithOptionName(model.contexts.defaultConnection.serverName);
-		} else {
-			if (model.contexts.defaultConnection) {
-				this.selectWithOptionName(model.contexts.defaultConnection.title ? model.contexts.defaultConnection.title : model.contexts.defaultConnection.serverName);
-			} else {
-				this.select(0);
-			}
-		}
-		otherConnections = this.setConnectionsList(model.contexts.defaultConnection, model.contexts.otherConnections);
-		let connections = otherConnections.map((context) => context.title ? context.title : context.serverName);
-		return connections;
-	}
-
-	private setConnectionsList(defaultConnection: ConnectionProfile, otherConnections: ConnectionProfile[]) {
-		if (defaultConnection.serverName !== msgSelectConnection) {
-			otherConnections = otherConnections.filter(conn => conn.id !== defaultConnection.id);
-			otherConnections.unshift(defaultConnection);
-			if (otherConnections.length > 1) {
-				otherConnections = otherConnections.filter(val => val.serverName !== msgSelectConnection);
-			}
-		}
-		return otherConnections;
-	}
-
-	public getSelectedConnection(selection: string): ConnectionProfile {
-		// Find all connections with the the same server as the selected option
-		let connections = this.model.contexts.otherConnections.filter((c) => selection === c.title);
-		// If only one connection exists with the same server name, use that one
-		if (connections.length === 1) {
-			return connections[0];
-		} else {
-			return find(this.model.contexts.otherConnections, (c) => selection === c.title);
+			this.enable();
 		}
 	}
 
 	public doChangeContext(connection?: ConnectionProfile, hideErrorMessage?: boolean): void {
-		if (this.value === msgAddNewConnection) {
+		if (this.value === msgChangeConnection || this.value === msgSelectConnection) {
 			this.openConnectionDialog();
```

Comment:
>openConnectionDialog [](start = 8, length = 20)

You don't want to prepopulate the dialog with the current connection when changing connection? Maybe they just want to switch DBs or something in which case that could save some typing